server-spi-private/src/main/java/org/keycloak/cookie/CookieType.java

```diff
@@ -5,11 +5,11 @@
 public final class CookieType {
 
     public static final CookieType AUTH_DETACHED = CookieType.create("KC_STATE_CHECKER")
-            .scope(CookieScope.LEGACY)
+            .scope(CookieScope.INTERNAL)
             .build();
 
     public static final CookieType AUTH_RESTART = CookieType.create("KC_RESTART")
-            .scope(CookieScope.LEGACY)
+            .scope(CookieScope.FEDERATION)
```

Comment:
This one needs to be available after a redirect. This is proven by https://github.com/keycloak/keycloak/issues/20490

Reply:
It is not, but if it's Strict it is not sent by the browser on a redirect

Reply:
I can't find any information about these cookies being omitted on redirects, do you have some documentation around that? Perhaps the issue is rather that the `Domain` or `Path` are simply not matching?

Reply:
It's basically what SameSite does as it's a CSRF protection mechanism

Reply:
It's basically what SameSite does as it's a CSRF protection mechanism

Reply:
> The strict value will prevent the cookie from being sent by the browser to the target site in all cross-site browsing contexts, even when following a regular link.

Ok, but this is not a cross-site cookie right? I really can't find anything about this not being preserved as long as the redirect is same site.

Reply:
Or wait, this is also the case for when users land from another website as a redirect? Ok, I see. That is the case here?

Reply:
Yup, so a regular OAuth authorization code request for instance is considered a `redirect`. Just to make matters even more fun (which is why `None` is required instead of `Lax`) with SAML you often do a POST binding, which is basically a cross-domain POST request (which is just awesome from a CSRF perspective, but hey).

Reply:
Well, that's just fantastic :laughing: Thanks for clarifying.